### PR TITLE
library: add screencast entry

### LINKED ISSUE #342: Add Screencast Demo
https://libportal.org/method.Portal.create_screencast_session.html

### PR DESCRIPTION
**Pull request by halfmexican**
closes #342

### CODE REVIEW

File: src/Library/demos/Screencast/main.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        const source = Gst.ElementFactory.make("pipewiresrc", "source");
+        const queue = Gst.ElementFactory.make("queue", "queue"); // add a queue element
+        const paintable_sink = Gst.ElementFactory.make(
+          "gtk4paintablesink",
+          "paintable_sink",
+        );
+        const glsinkbin = Gst.ElementFactory.make("glsinkbin", "glsinkbin");
+
+        // Set up and Link Pipeline
+        source.set_property("fd", pw_remote); // pw_remote is the file descriptor obtained from libportal
```
Comment (bilelmoussaoui) [inline src/Library/demos/Screencast/main.js:62]: you should also set the node id but no idea how you would get that with libportal

Comment (sonnyp) [reply src/Library/demos/Screencast/main.js:62]: Thanks, yeah that's also what Georges said, I was going to look into this

Comment (sonnyp) [reply src/Library/demos/Screencast/main.js:62]: @halfmexican you can find the node id from https://libportal.org/method.Session.get_streams.html

**Review by sonnyp — CHANGES_REQUESTED** [review]
It doesn't show the Camera stream anymore :)
But now we need to set the node id on the pipewire source as hinted by Bilal and Georges

https://libportal.org/method.Session.get_streams.html

@halfmexican also please use Gio._Promisify

File: src/Library/demos/Screencast/main.js (hunk 1 above)
Comment (halfmexican) [reply src/Library/demos/Screencast/main.js:62]: Think I did it

**Review by sonnyp — APPROVED** [review]
Good job.

I refactored the JS code to use promisify correctly https://github.com/sonnyp/Workbench/pull/377/commits/f6e2e894e6de6b7a1436d8a7da1b69dde18026d6

**Comment by sonnyp** [comment]
Not merging this just yet as it doesn't work with GNOME 45 for some reason. I need to do more investigation.

**Pull request merged**